projects/caddy/build.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -eu
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+mv $SRC/parser_fuzzer.go $SRC/caddy/caddyconfig/caddyfile/
+
+if [[ $SANITIZER = *coverage* ]]; then
+	go get github.com/caddyserver/caddy/caddyconfig/caddyfile
+	compile_go_fuzzer github.com/caddyserver/caddy/caddyconfig/caddyfile FuzzParser parser_fuzzer
+
+	compile_go_fuzzer github.com/caddyserver/caddy FuzzParseNetworkAddress fuzz_parse_network_address
+
+	# Coverage is not supported for this fuzzer:
+	#compile_go_fuzzer github.com/caddyserver/caddy/caddyconfig/httpcaddyfile FuzzParseAddress fuzz_parse_address
+	compile_go_fuzzer github.com/caddyserver/caddy FuzzReplacer fuzz_replacer
+	exit 0
+fi
+
+compile_go_fuzzer ./caddyconfig/caddyfile FuzzParser parser_fuzzer
+compile_go_fuzzer ./ FuzzParseNetworkAddress fuzz_parse_network_address
+compile_go_fuzzer ./caddyconfig/httpcaddyfile FuzzParseAddress fuzz_parse_address
+compile_go_fuzzer ./ FuzzReplacer fuzz_replacer
```

Comment:
Adopted from #5624, how about we approach it in a more general way to automatically accommodate newer fuzzers as they're added upstream?

I haven't been able to get the coverage sanitizer to play well with Go's SIV. It's probably best to disable it for now. N.B. the package path `github.com/caddyserver/caddy/caddyconfig/caddyfile` is for Caddy v1.

```suggestion
if [[ $SANITIZER = *coverage* ]]; then
    # we don't support 'coverage' sanitizer yet because 
    # compile_go_fuzzer doesn't seem to like SIV.
    # TODO: investigate proper fix for either
    exit 0
fi

cd "$GOPATH"/src/github.com/caddyserver/caddy/v2

find . -name '*_fuzz.go' | while read -r target
do
# Arguments are :
#     path of the package with the fuzz target
#     name of the fuzz function
#     name of the fuzzer to be built
#     optional tag to be used by go build and such

    fuzzed_func=$(grep -o "Fuzz[a-zA-Z]\+" "$target")
    fuzzer_name=$(echo "$fuzzed_func" | sed -E 's/([A-Z])/-\L\1/g' | sed 's/^-//')
    # find the relative directory and remove the first `.` (`./` is removed for root)
    target_dir=$(dirname "$target"); target_dir="${target_dir//.}";
    target_corpus_name="${fuzzer_name}_seed_corpus.zip"

    curl -s -f -O "https://raw.githubusercontent.com/caddyserver/caddy/fuzz-seed-corpus/${target_corpus_name}" || true
    compile_go_fuzzer github.com/caddyserver/caddy/v2"$target_dir" "$fuzzed_func" "$fuzzer_name" gofuzz
done
```